doctrine.rst

```diff
@@ -721,155 +520,171 @@ method of the entity manager::
-like to remove the given object from the database. The actual ``DELETE`` query,
-however, isn't actually executed until the ``flush()`` method is called.
+like to remove the given object from the database. The ``DELETE`` query isn't
+actually executed until the ``flush()`` method is called.
 
 .. _doctrine-queries:
 
-Querying for Objects
---------------------
+Querying for Objects: The Repository
+------------------------------------
 
 You've already seen how the repository object allows you to run basic queries
 without any work::
 
+    // from inside a controller
     $repository = $this->getDoctrine()->getRepository(Product::class);
 
-    $product = $repository->find($productId);
-    $product = $repository->findOneByName('Keyboard');
+    $product = $repository->find($id);
 
-Of course, Doctrine also allows you to write more complex queries using the
-Doctrine Query Language (DQL). DQL is similar to SQL except that you should
-imagine that you're querying for one or more objects of an entity class (e.g. ``Product``)
-instead of querying for rows on a table (e.g. ``product``).
+But what if you need a more complex query? When you generated your entity with
+``make:entity``, the command *also* generated a ``ProductRepository`` class::
 
-When querying in Doctrine, you have two main options: writing pure DQL queries
-or using Doctrine's Query Builder.
+    // src/Repository/ProductRepository.php
+    namespace App\Repository;
 
-Querying for Objects with DQL
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    use App\Entity\Product;
+    use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
 
-Imagine that you want to query for products that cost more than ``19.99``,
-ordered from least to most expensive. You can use DQL, Doctrine's native
-SQL-like language, to construct a query for this scenario::
+    class ProductRepository extends ServiceEntityRepository
+    {
+    }
 
-    $query = $em->createQuery(
-        'SELECT p
-        FROM App:Product p
-        WHERE p.price > :price
-        ORDER BY p.price ASC'
-    )->setParameter('price', 19.99);
+When you fetch your repository (i.e. ``->getRepository(Product::class)``, it is
+*actually* an instance of *this* object! This is because of the ``repositoryClass``
+config that was generated at the top of your ``Product`` entity class.
 
-    $products = $query->getResult();
+Suppose you want to query for all Product objects greater than a certain price. Add
+a new method for this to your repository::
 
-If you're comfortable with SQL, then DQL should feel very natural. The biggest
-difference is that you need to think in terms of selecting PHP objects,
-instead of rows in a database. For this reason, you select *from* the
-``App:Product`` *entity* (an optional shortcut for the
-``App\Entity\Product`` class) and then alias it as ``p``.
+    // src/Repository/ProductRepository.php
 
-.. tip::
+    // ...
+    class ProductRepository extends ServiceEntityRepository
```

Comment:
if we are extending from `ServiceEntityRepository` shouldn't assume that this class is intended to be a service and autowired? if yes, we might need the `__constructor` to set the entityClass?